src/tasks/audio.c

```diff
@@ -65,7 +65,9 @@ int do_task_master_core(struct sof *sof)
 	sys_comp_mixer_init();
 	sys_comp_mux_init();
 	sys_comp_switch_init();
+#ifdef CONFIG_VOLUME
 	sys_comp_volume_init();
+#endif
```

Comment:
@dbaluta I think its not a good idea in this case. The point of this PR is to not include source files of disabled modules so I cannot expand source of sys_comp_volume_init().

Also in the same file a few lines below there is init_static_pipeline() used with the similar ifdefs so its already a common way.

Reply:
@abonislawski you will not any source files of the disabled modules. The empty comp_init() will be made inline inside a header file while the real comp_init() will be made available when the CONFIG_VOLUME is set.

Reply:
@dbaluta you are right, I can do that easily if needed. Lets wait for more reviewers.

I have also changed commit subjects as you requested

Reply:
> @dbaluta you are right, I can do that easily if needed. Lets wait for more reviewers.
> 
> I have also changed commit subjects as you requested

@abonislawski  Sure, thanks a lot for doing this.

Minors nits if you need to send a new version.
- capitalize the subject line
- Use the imperative mood in the subject line (patch 3 doesn't even contain a verb in the subject)
See more details here [1]

But these are minors things. Lets wait for others review.

[1] https://chris.beams.io/posts/git-commit/#seven-rules

Reply:
@dbaluta thanks, all 3 commit names are okay now I hope :)

Reply:
@abonislawski did you make the static inline change as @dbaluta suggested. I cant see it in the latest push.

Reply:
@lrgirdwod @dbaluta @abonislawski How about using weak symbols for initializers and make a call based on their non-null test at runtime?

Reply:
@mmaka1 I think the best option would be as I suggested using static inline function. When the symbol CONFIG_COMP_XXX is not defined the compiler will likely compile out the empty static inline function definition anyway.

This practice is pretty common inside the Linux kernel. I think your proposal using weak symbols is equivalent.

I don't have a strong opinion for any of them, it is your call on this or Liam's.

Reply:
I assumed a new component development process to be quite independent from the core/infrastructure code. Therefore keeping a single central list in the core header probably should be avoided, so that initializers declarations to be distributed to the components' public/api headers. And from compilation->linking->runtime stages perspective, the references to be maintained in the latest possible stage (ultimately querying and self-registration, like a plug-in/extension code, at run-time). Therefore moving to the linking from compilation was considered to be a bit better as a short term solution by me.

Reply:
@abonislawski As I said before either of the solutions are fine with me. You know better the plans for future extending the component part.

I think @lgirdwood will have the final word.

Me, personally, I really don't understand what is the advantage of using the weak symbols solutions vs using an inline function.

Reply:
Option two from @jajanusz sounds nicer to me. Though that's my feel as mostly a component developer without e.g. that control of system aspect. I always forget to edit init_components() when adding a component.

Reply:
@jajanusz @singalsu thats right, option two, self registering components, was already discussed and I think this will be done sooner or later but for sure in another PR. In this PR we are discussing about temporary solution because the scope of this PR is different. So, we have 3 options (which will be changed anyway sooner or later):
1. Ifdefs like now to call or not to call a function
2. inline empty funcs like @dbaluta proposed
3. weak symbols like @mmaka1 proposed (maybe with aliases?)

Reply:
This also needs revisited  anyway in the future when runtime modules are supported, since we will be adding and removing modules at runtime. The __attribute__((constructor)) and __attribute((destructor)) will support modules so should be used. However if time does not permit atm for attributes, then static inline will do today.

Reply:
@lgirdwood its done.